src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -200,7 +203,11 @@ internal bool IsTaskOfType([NotNullWhen(returnValue: true)] ITypeSymbol? typeSym
         {
             return typeSymbol != null
                 && typeSymbol.OriginalDefinition != null
-                && typeSymbol.OriginalDefinition.Equals(GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask1))
+                &&
+#if CODEANALYSIS_V3_OR_BETTER
+                    SymbolEqualityComparer.Default.
+#endif
+                    Equals(typeSymbol.OriginalDefinition, GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask1))
```

Comment:
Same concern about performance if the extension is called multiple times in the same analyzer.